app/src/main/java/de/danoeh/antennapod/dialog/FilterDialog.java

```diff
@@ -21,39 +28,103 @@ public FilterDialog(Context context, FeedItemFilter feedItemFilter) {
         this.filter = feedItemFilter;
     }
 
-    public void openDialog() {
-        final String[] items = context.getResources().getStringArray(R.array.episode_filter_options);
-        final String[] values = context.getResources().getStringArray(R.array.episode_filter_values);
-        final boolean[] checkedItems = new boolean[items.length];
+    public void openCustomDialog() {
 
         final Set<String> filterValues = new HashSet<>(Arrays.asList(filter.getValues()));
+        AlertDialog.Builder builder = new AlertDialog.Builder(context);
+        builder.setTitle(R.string.filter);
 
-        // make sure we have no empty strings in the filter list
-        for (String filterValue : filterValues) {
-            if (TextUtils.isEmpty(filterValue)) {
-                filterValues.remove(filterValue);
-            }
-        }
+        LayoutInflater inflater = LayoutInflater.from(this.context);
+        LinearLayout layout = (LinearLayout) inflater.inflate(R.layout.filter_dialog_layout, null, false);
+        builder.setView(layout);
+
+        for (FeedItemFilterGroup.FeedItemEnum item : FeedItemFilterGroup.FeedItemEnum.values()) {
 
-        for (int i = 0; i < values.length; i++) {
-            String value = values[i];
-            if (filterValues.contains(value)) {
-                checkedItems[i] = true;
+            RelativeLayout row = (RelativeLayout) inflater.inflate(R.layout.filter_dialog_relative_cardview, null);
+            RadioButton radioButton1 = row.findViewById(R.id.filter_dialog_radioButton1);
+            RadioButton radioButton2 = row.findViewById(R.id.filter_dialog_radioButton2);
+            RadioButton radioButton3 = row.findViewById(R.id.filter_dialog_radioButton3);
+            radioButton1.setText(item.values[1].displayName);
+            radioButton1.setTextColor(Color.BLACK);
+            radioButton2.setText(item.values[0].displayName);
+            radioButton2.setTextColor(Color.BLACK);
+
+            Iterator<String> filterIterator = filterValues.iterator();
+            while (filterIterator.hasNext()) {
+                String nextItem = filterIterator.next();
+                if (item.values[1].filterId.equals(nextItem)) {
+                    item.values[1].setSelected(true);
+                    item.values[0].setSelected(false);
+                    radioButton1.setBackgroundResource(R.color.accent_light);
+                    radioButton2.setBackgroundResource(R.color.master_switch_background_light);
+                    radioButton3.setBackgroundResource(R.drawable.filter_dialog_x_on);
+                    radioButton1.setSelected(true);
+                    radioButton2.setSelected(false);
+                    radioButton1.setTextColor(Color.WHITE);
+                    radioButton2.setTextColor(Color.BLACK);
```

Comment:
I do not really like this duplicated code. Things like this are always hard to maintain.

> One thing; no selectors because every time an element is clicked it needs to update another element, if you can do that with a radio group I still need to update the clear image drawable and it seems to make more sense just to use a couple lines of code where everything else is.

If you make the clear button a RadioButton and move the RadioGroup out of the CardView, so that it contains the clear button, you do not have to care about de-selecting other items and changing their background. This can then be done automatically.

Reply:
right

Reply:
Here is the problem:

You want rounded corners with no graphics. The clear button was always a radioButton because I've tried to make that work. It doesn't.

The left and right radioButtons have to be nested in the cardView so the cardView can handle giving the rounded corner effect on the radioButtons. The only way I can have 3 is if I do something like this -->

( left  | middle | right )

doing...

( left | right ) (clear) would require 2 cardViews, more confusion less maintainable right?

So I can only fit a radioGroup with 2 radioButtons to give the desired effect that you want.
To keep the effect you want I had to hardcode the setSelected because I need the clear button outside the radioGroup and if I'm going to keep it outside the radioGroup because I have to I can't use a selection because each element changes another updates another element outside the selection.

If you want I can put the drawables back and keep everything in 1 radioGroup but you can't have both and still get the rounded corner overlay on the radioButton colors.

Reply:
Doesn't it work to put the RadioGroup around both, the CardView and the Clear button? Then all buttons are in one RadioGroup and only two of them are in the CardView.

```
<RadioGroup>
    <CardView>
        <LinearLayout>
            <RadioButton />
            <RadioButton />
            ...
            <RadioButton />
        </LinearLayout>
    </CardView>
    <RadioButton /> <!-- Clear button -->
</RadioGroup>
```

Reply:
Ooh wait. Now I get the problem. RadioButtons only work that way if they are direct children of the RadioGroup. I will try to find a solution for this.

Reply:
Okay, another thing I could do is just have the 2 on/off in the radioGroup and selector and only update 1 thing(clear button) in the code onclick event() to simplify it more.